### PR TITLE
Fix handling of start and end time in legacy

### LINKED ISSUE #1367: Better handling of dates in Qt (like job start_date and end_date)
Part of #566

### PR DESCRIPTION
**Pull request by kvashchuka**
**Issue**
Resolves #1367 

**Approach**
Change format of start and end time in snapshot to datetime. Custom data marshaller and un-marshaller were added to convert to and from json

### CODE REVIEW

**Comment by sondreso** [comment]
I think a rebase on current refactor-branch would be nice 😄

File: ert_shared/status/tracker/legacy.py [hunk 1]
```diff
@@ -175,8 +175,8 @@ def _create_snapshot_dict(
                 job = step.jobs[str(idx)]
 
                 # FIXME: parse these as iso date
```
Comment (jondequinor) [inline ert_shared/status/tracker/legacy.py:177]: Remove the fixme—you've fixed it :joy:

**Review by jondequinor — APPROVED** [review]
A part from the one comment, I think this is very good! 👏

File: ert_shared/ensemble_evaluator/entity/ensemble_base.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import asyncio
 import websockets
 
+import ert_shared.ensemble_evaluator.entity.serialization as serialization
```
Comment (jondequinor) [inline ert_shared/ensemble_evaluator/entity/ensemble_base.py:4]: This is probably better written as 
`from ert_shared.ensemble_evaluator.entity import serialization` ? Idk

File: tests/ensemble_evaluator/test_data_marshaller_and_unmarsaller.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import datetime
+
+from cloudevents.http import from_json, to_json
+from cloudevents.http.event import CloudEvent
+
+from ert_shared.ensemble_evaluator.entity import serialization
+
+
+def test_data_marshaller_and_unmarshaller(ee_id=0, event_index=0):
```
Comment (jondequinor) [inline tests/ensemble_evaluator/test_data_marshaller_and_unmarsaller.py:9]: Not sure what those parameters mean? I am unfamiliar with that approach to writing tests.

Comment (jondequinor) [inline tests/ensemble_evaluator/test_data_marshaller_and_unmarsaller.py:9]:
```suggestion
def test_data_marshaller_and_unmarshaller():
    ee_id = 0
    event_index = 0
```

Comment (kvashchuka) [reply tests/ensemble_evaluator/test_data_marshaller_and_unmarsaller.py:9]: my bad, we don't need this parameters at all

**Comment by jondequinor** [comment]
Excellent. Merge this when the tests passes =)

**Pull request merged**